internal/pkg/manifest/workload_ecs.go

```diff
@@ -85,13 +85,57 @@ type ImageWithPortAndHealthcheck struct {
 	HealthCheck   ContainerHealthCheck `yaml:"healthcheck"`
 }
 
+// AlarmArgs represents specs of CloudWatch alarms for deployment rollbacks.
+type AlarmArgs struct {
+	CPUUtilization    *int `yaml:"cpu_utilization"`
+	MemoryUtilization *int `yaml:"memory_utilization"`
+}
+
+// RollbackAlarmArgsOrNames represents alarms for rolling back deployments.
+// It supports unmarshalling yaml which can either be of type AlarmArgs or a list of strings.
+type RollbackAlarmArgsOrNames struct {
+	AlarmNames []*string
+	AlarmArgs
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the RollbackAlarmArgsOrNames
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v3) interface.
+func (r *RollbackAlarmArgsOrNames) UnmarshalYAML(value *yaml.Node) error {
+	if err := value.Decode(&r.AlarmArgs); err != nil {
+		var yamlTypeErr *yaml.TypeError
+		if !errors.As(err, &yamlTypeErr) {
+			return err
+		}
+	}
+	if !r.AlarmArgs.isEmpty() {
+		// Unmarshaled successfully to r.AlarmArgs, unset r.AlarmNames, and return.
+		r.AlarmNames = nil
+		return nil
+	}
+	if err := value.Decode(&r.AlarmNames); err != nil {
+		return errUnmarshalRollbackAlarms
+	}
+	return nil
+}
+
+// IsEmpty returns true if neither alarm names nor alarm args (for creating alarms) are present.
+func (r *RollbackAlarmArgsOrNames) IsEmpty() bool {
+	return r.AlarmNames == nil && r.AlarmArgs.isEmpty()
+}
+
+func (a *AlarmArgs) isEmpty() bool {
+	return a.CPUUtilization == nil && a.MemoryUtilization == nil
+}
+
 // DeploymentConfiguration represents the deployment strategies for a service.
 type DeploymentConfiguration struct {
-	Rolling *string `yaml:"rolling"`
+	Rolling *string                  `yaml:"rolling"`
+	Alarms  RollbackAlarmArgsOrNames `yaml:"rollback_alarms"`
 }
 
 func (d *DeploymentConfiguration) isEmpty() bool {
-	return d == nil || d.Rolling == nil
+	return d.Rolling == nil && d.Alarms.IsEmpty()
```

Comment:
Can we make sure that whoever calls `d.isEmpty()` is not counting on `isEmpty()` to determine whether `d==nil`? That is, I'm worried that we have codes like this:
```go
// d is a (*DeploymentConfiguration)(nil)
if !d.isEmpty() {}
```

This code wouldn't panic before - `d.isEmpty()` would have evaluated `d==nil` and return `true`. Now it would panic, because `d.isEmpty` will try to evaluate `d.Rolling` which is fetching data from a `nil` pointer.